packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,221 @@
+			if (selectedDateTime !== undefined) {
+				const date = selectedDateTime;
+				const hours = getHours(selectedDateTime);
+				const minutes = getMinutes(selectedDateTime);
+				const time = hoursAndMinutesToTime(hours, minutes);
+				return {
+					date,
+					time,
+					textInput: getTextDate(date, time),
+				};
+			}
+
+			return {
+				textInput: '',
+			};
+		})();
+
+		this.onChangeInput = this.onChangeInput.bind(this);
+		this.onSubmitPicker = this.onSubmitPicker.bind(this);
+	}
+
+	onSubmitPicker({ date, time }) {
+		this.updateDateTime(date, time);
+	}
+
+	onChangeInput(event) {
+		const text = event.target.value;
+
+		const splitMatches = text.match(splitDateAndTimeRegex);
+
+		if (!splitMatches) {
+			console.warn('DATETIME - INCORRECT FORMAT:', text);
+			this.updateDateTime(undefined, undefined, text);
+			return;
+		}
+
+		const [
+			,
+			dateText,
+			timeText,
+		] = splitMatches;
+
+		const date = (() => {
+			const dateMatches = dateText.match(dateRegex);
+			if (!dateMatches) {
+				console.warn('DATE - INCORRECT FORMAT:', dateText);
+				return undefined;
+			}
+
+			const [
+				,
+				dayString,
+				monthString,
+				yearString,
+			] = dateMatches;
+
+			const day = parseInt(dayString, 10);
+			const month = parseInt(monthString, 10);
+			const monthIndex = month - 1;
+			const year = parseInt(yearString, 10);
+
+			if (month === 0 || month > 12) {
+				console.warn('DATE - INCORRECT MONTH NUMBER:', dateText);
+				return undefined;
+			}
+
+			if (day === 0) {
+				console.warn('DATE - INCORRECT DAY NUMBER:', dateText);
+				return undefined;
+			}
+
+			const monthDate = new Date(year, monthIndex);
+			const lastDateOfMonth = lastDayOfMonth(monthDate);
+
+			if (day > getDate(lastDateOfMonth)) {
+				console.warn('DATE - INCORRECT DAY NUMBER RELATIVE TO MONTH:', dateText);
+				return undefined;
+			}
+
+			const dateValidated = setDate(monthDate, day);
+
+			return dateValidated;
+		})();
+
+
+		const time = (() => {
+			const timeMatches = timeText.match(timeRegex);
+			if (!timeMatches) {
+				console.warn('TIME - INCORRECT FORMAT:', timeText);
+				return undefined;
+			}
+
+			const [
+				,
+				hoursString,
+				minutesString,
+			] = timeMatches;
+
+			const hours = parseInt(hoursString, 10);
+
+			if (hours >= 24) {
+				console.warn('TIME - INCORRECT HOUR NUMBER:', timeText);
+				return undefined;
+			}
+
+			const minutes = parseInt(minutesString, 10);
+
+			if (minutes >= 60) {
+				console.warn('TIME - INCORRECT MINUTES NUMBER:', timeText);
+				return undefined;
+			}
+
+			return hoursAndMinutesToTime(hours, minutes);
+		})();
+
+		this.updateDateTime(date, time, text);
+	}
+
+	updateDateTime(date, time, textInput = getTextDate(date, time)) {
+		this.setState({
+			date,
+			time,
+			textInput,
+		});
+
+		const fullDate = (() => {
+			if (date === undefined || time === undefined) {
+				return undefined;
+			}
+
+			return setMinutes(date, time);
+		})();
+		this.props.onChange(fullDate);
```

Comment:
As said in the proptypes, onChange should be called with text.
The exchange format with the outside world is text as you implement a form input. Dates are just internal formats to implement the logic.

+ it will remove the (() => {})()

Reply:
For later, it's have no impact right now
✘